Default to zero byte reads when we use PipeReader.Create

davidfowl opened the pull request:
- This will avoid allocating buffers until there's data available to read.

Fixes https://github.com/dotnet/aspnetcore/issues/42829

davidfowl committed:
Default to zero byte reads when we use PipeReader.Create
- This will avoid allocating buffers until there's data available to read.

JamesNK commented:
Unit tests? I know there are tests for the stream adapter using zero byte reads, but I'd like some unit tests in asp.net core to ensure it never regresses.

davidfowl commented:
It's not worth it. Ideally I would be able to verify that the option is set in all of these cases but it's really not worth setting up scaffolding to verify that the option works.

davidfowl committed:
Undo the client side changes for signalr
- Riskier since the client runs on older .NET versions that may not support zero byte reads

davidfowl committed:
Add test for zero byte body reads

davidfowl committed:
Added zero byte reads for the request body in kestrel.

davidfowl committed:
Ignore logging zero byte reads

davidfowl commented:
/azp run

halter73 reviewed (approved):
It'd be nice to get benchmarks, but processing request with large bodies is usually bandwidth constrained. This is all the more reason to save memory.
